src/common/png_image.cpp

```diff
@@ -97,31 +104,33 @@ PNGReader::PNGReader(std::filesystem::path image_path, Logger& logger)
     png_read_image(png, row_pointers.get());
 
     logger.info("Picture successfully open with path: {}", image_path.string());
-    fclose(fp);
-
-    png_destroy_read_struct(&png, &info, NULL);
+    onClose();
+    m_empty = false;
   }
   else {
     logger.error("Can't open file with path: {}", image_path.string());
   }
 }
 
 
-std::unique_ptr<std::uint8_t[]> PNGReader::get_data() {
+std::unique_ptr<std::uint8_t[]> PNGImage::extract_data() {
   return std::move(m_data);
 }
 
-usize PNGReader::get_width() {
+usize PNGImage::get_width() {
   return m_width;
 }
 
-usize PNGReader::get_height() {
+usize PNGImage::get_height() {
   return m_height;
 }
 
-usize PNGReader::get_channels() {
+usize PNGImage::get_channels() {
   return m_channels;
 }
 
+bool PNGImage::empty() {
+  return m_empty;
```

Comment:
I would expect that `empty()` returns `true`  only in case if `width * height == 0`. There should be a better name for this method.